src/Illuminate/Collections/LazyCollection.php

```diff
@@ -1010,6 +1012,51 @@ public function split($numberOfGroups)
         return $this->passthru('split', func_get_args());
     }
 
+    /**
+     * Get the first item in the collection, but only if exactly
+     * item exists. Otherwise, throw an exception.
+     *
+     * @param  callable|null  $callback
+     * @return mixed
+     *
+     * @throws ItemNotFoundException
+     * @throws MultipleItemsFoundException
+     */
+    public function sole(callable $callback = null)
```

Comment:
This implementation is way more complex than it needs to be.

Something like this (not tested) should suffice:

```php
public function sole(callable $callback = null)
{
    $items = $this
        ->when($callback)
        ->filter($callback)
        ->take(2)
        ->collect();

    if ($items->isEmpty()) {
        throw new ItemNotFoundException;
    }

    if (! $items->containsOneItem()) {
        throw new MultipleItemsFoundException;
    }

    return $items[0];
}
```

Reply:
I do agree that it's more complex. I was looking at some of the existing code for things like the ` first() ` method and just assumed that it had been done like that for a specific reason. I've not really used LazyCollections too much so I wasn't 100% sure on the best way to implement it. Thanks for the advice on this one